src/parcsr_ls/par_mgr.c

```diff
@@ -247,6 +247,20 @@ hypre_MGRDestroy( void *data )
          hypre_ParVectorDestroy((mgr_data -> F_array)[i]);
          hypre_ParVectorDestroy((mgr_data -> U_array)[i]);
 
+         if ((mgr_data -> R_array)[i - 1] && (mgr_data -> RT_array)[i - 1] &&
+             ((mgr_data -> R_array)[i - 1] == (mgr_data -> RT_array)[i - 1]))
+         {
+            hypre_ParCSRMatrixDestroy((mgr_data -> R_array)[i - 1]);
+            (mgr_data -> R_array)[i - 1] = NULL;
+         }
+
+         if ((mgr_data -> P_array)[i - 1] && (mgr_data -> RT_array)[i - 1] &&
+             ((mgr_data -> P_array)[i - 1] == (mgr_data -> RT_array)[i - 1]))
+         {
+            hypre_ParCSRMatrixDestroy((mgr_data -> P_array)[i - 1]);
+            (mgr_data -> P_array)[i - 1] = NULL;
+         }
+
```

Comment:
@victorapm a few comments here.
* Could we get away with just checking for "=="? Are the first two checks in the if-statement needed?
* Do we have currently options where we set RT_array[j] = R_array[j]?
* Similarly for RT_array[j] = P_arrayj] (I have support for this in the sMGR branch based on our previous discussions, but it has not been merged into master yet)

Reply:
1. I suppose the goal is to avoid potential double free error when RT == R (is this correct?). Just to clarify, is the first two checks meant to address the case when both could be NULL? Is that what you meant by your example?
2. Yes if you can't think of a case where we would need that, I agree we should remove that.
3. Sure, that is good with me
4. This is mainly for completeness since the data it points to has been deleted. So similar to what you have in (line 418 of par_mgr_setup.c (or see code snippet in comment below)

Reply:
Oh for 4. I was just pointing out that we should probably add:
(mgr_data -> RT_array)[i - 1] = NULL;
after
(mgr_data -> P_array)[i - 1] = NULL;